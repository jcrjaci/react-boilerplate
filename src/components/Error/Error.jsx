import React from 'react';
import PropTypes from 'prop-types';
import './Error.scss';

const Error = ({ msg }) => (
  <div className="error-msg">
    <span>{msg || 'An error occurred'}</span>
  </div>
);

Error.propTypes = {
  msg: PropTypes.string,
};

Error.defaultProps = {
  msg: 'An error occurred',
};

export default Error;
