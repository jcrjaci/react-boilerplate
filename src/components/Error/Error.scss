.error-msg {
  background-color: red;
  width: 100%;
}