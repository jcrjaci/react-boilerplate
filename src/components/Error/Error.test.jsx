import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Error from './Error';

configure({ adapter: new Adapter() });

describe('<Error />', () => {
  const wrapper = shallow(<Error msg="Oops something went wrong." />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render default msg prop in case of falsy value', () => {
    wrapper.setProps({ msg: ''});
    expect(wrapper.find('.error-msg').children().text()).toBe('An error occurred');

    wrapper.setProps({ msg: undefined });
    expect(wrapper.find('.error-msg').children().text()).toBe('An error occurred');
  });
});
