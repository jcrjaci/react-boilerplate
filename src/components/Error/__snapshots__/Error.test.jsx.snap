// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly 1`] = `
<div
  className="error-msg"
>
  <span>
    Oops something went wrong.
  </span>
</div>
`;
